### PR TITLE
Prevent firelog from logging invalid data

### PR DESCRIPTION
**Pull request by chliangGoogle**

### CODE REVIEW

File: FirebaseMessaging/Sources/FIRMessagingExtensionHelper.m [hunk 1]
```diff
@@ -211,8 +211,25 @@ - (void)exportDeliveryMetricsToBigQueryWithMessageInfo:(NSDictionary *)info {
...
   fm_MessagingClientEvent clientEvent = fm_MessagingClientEvent_init_default;
+  if (!info[kFIRMessagingSenderID]) {
+    FIRMessagingLoggerError(kFIRMessagingServiceExtensionInvalidProjectID,
+                            @"Delivery logging failed: Invalid project ID");
+    return;
+  }
   clientEvent.project_number = (int64_t)[info[kFIRMessagingSenderID] longLongValue];
+
+  if (!info[kFIRMessagingMessageIDKey]) {
+    FIRMessagingLoggerWarn(kFIRMessagingServiceExtensionInvalidMessageID,
```
Comment (maksymmalyhin) [inline FirebaseMessaging/Sources/FIRMessagingExtensionHelper.m:222]: Seems like this code path is not covered by tests. I would suggest adding a test/tests for each condition.

File: FirebaseMessaging/Sources/FIRMessagingExtensionHelper.m [hunk 2]
```diff
@@ -211,8 +211,25 @@ - (void)exportDeliveryMetricsToBigQueryWithMessageInfo:(NSDictionary *)info {
   fm_MessagingClientEventExtension eventExtension = fm_MessagingClientEventExtension_init_default;
 
   fm_MessagingClientEvent clientEvent = fm_MessagingClientEvent_init_default;
+  if (!info[kFIRMessagingSenderID]) {
```
Comment (maksymmalyhin) [inline FirebaseMessaging/Sources/FIRMessagingExtensionHelper.m:214]: optional: I usually prefer using a local var in such cases like:
```
NSString *senderID = info[kFIRMessagingSenderID];
if(senderID.length == 0) {
...
}
clientEvent.project_number = (int64_t)[info[kFIRMessagingSenderID] longLongValue];
```

It makes relation between validation and usage more clear.

**Review by maksymmalyhin — APPROVED** [review]
LGTM with a couple minor comments

**Review by maksymmalyhin — APPROVED** [review]

File: FirebaseMessaging/Sources/FIRMessagingExtensionHelper.m (hunk 2 above)
Comment (chliangGoogle) [reply FirebaseMessaging/Sources/FIRMessagingExtensionHelper.m:214]: what if it's not a string? I worry it might still pass and crash the protobuf.

Comment (maksymmalyhin) [reply FirebaseMessaging/Sources/FIRMessagingExtensionHelper.m:214]: The validation is just an example, my main point here is that declaring a local variable is more readable and change proof than accessing a value from the dict by key. I started doing it after I had changed a key in one place but forgot to change at another by accident.

File: FirebaseMessaging/Sources/FIRMessagingExtensionHelper.m (hunk 1 above)
Comment (chliangGoogle) [reply FirebaseMessaging/Sources/FIRMessagingExtensionHelper.m:223]: added tests

File: FirebaseMessaging/Sources/FIRMessagingExtensionHelper.m [hunk 3]
```diff
@@ -211,8 +211,26 @@ - (void)exportDeliveryMetricsToBigQueryWithMessageInfo:(NSDictionary *)info {
...
+    return;
+  }
   clientEvent.project_number = (int64_t)[info[kFIRMessagingSenderID] longLongValue];
+
+  if (!info[kFIRMessagingMessageIDKey] ||
+      ![info[kFIRMessagingMessageIDKey] isKindOfClass:NSString.class]) {
+    FIRMessagingLoggerWarn(kFIRMessagingServiceExtensionInvalidMessageID,
+                           @"Delivery logging failed: Invalid Message ID");
+    return;
+  }
```
[the remark this answers is not in the document]
Comment (chliangGoogle) [reply FirebaseMessaging/Sources/FIRMessagingExtensionHelper.m:226]: Done.

**Pull request merged**